### PR TITLE
Add support for querying data models through graphql

### PR DESCRIPTION
**Pull request by erlendvollset**
- Add support for querying through graphql
- Bump version and update changelog
- Update docs

## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/data_modeling/graphql.py [hunk 1]
```diff
@@ -133,3 +133,29 @@ def apply_dml(
...
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.data_modeling.graphql.query(
+                ...     id=("mySpace", "myDataModel", "v1"),
+                ...     query="listThings { items { thingProperty } }",
+                ... )
+        """
+        dm_id = DataModelId.load(id)
+        endpoint = f"/userapis/spaces/{dm_id.space}/datamodels/{dm_id.external_id}/versions/{dm_id.version}/graphql"
```
Comment (haakonvt) [inline cognite/client/_api/data_modeling/graphql.py:159]: Should we need to url encode this?

Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:160]: Absolutely. Fixed. I wrongly assumed APIClient/HttpClient would do that for us. (We should probably make it so).

File: cognite/client/_api/data_modeling/graphql.py [hunk 2]
```diff
@@ -133,3 +134,32 @@ def apply_dml(
...
+                ...     id=("mySpace", "myDataModel", "v1"),
+                ...     query="listThings { items { thingProperty } }",
+                ... )
+        """
+        dm_id = DataModelId.load(id)
+        endpoint = interpolate_and_url_encode(
+            "/userapis/spaces/{}/datamodels/{}/versions/{}/graphql", dm_id.space, dm_id.external_id, dm_id.version
+        )
+        res = self._post_graphql(url_path=endpoint, query_name="", json={"query": query, "variables": variables})
+        return res
```
Comment (haakonvt) [inline cognite/client/_api/data_modeling/graphql.py:165]: If you want to golf 😅 
```suggestion
        return self._post_graphql(url_path=endpoint, query_name="", json={"query": query, "variables": variables})
```

File: cognite/client/_api/data_modeling/graphql.py [hunk 3]
```diff
@@ -133,3 +134,32 @@ def apply_dml(
         query_name = "upsertGraphQlDmlVersion"
         res = self._post_graphql(url_path="/dml/graphql", query_name=query_name, json=payload)
         return DMLApplyResult.load(res[query_name]["result"])
+
+    def query(self, id: DataModelIdentifier, query: str, variables: dict[str, Any] | None = None) -> dict[str, Any]:
+        """Apply the DML for a given data model.
```
Comment (haakonvt) [inline cognite/client/_api/data_modeling/graphql.py:139]: A bit cryptic for me at least what this means "Apply the DML for a given data model."

**Review by haakonvt — APPROVED** [review]
Lgtm! Just a last few comments/questions.

File: cognite/client/_api/data_modeling/graphql.py (hunk 3 above)
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:139]: Copy-paste blunder from the above method 😵‍💫

File: cognite/client/_api/data_modeling/graphql.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:165]: I actually prefer it as is - gives a nice place to place a breakpoint!

**Pull request merged**